actionpack/lib/action_dispatch/journey/gtg/simulator.rb

```diff
@@ -14,6 +14,8 @@ def initialize(memos)
       end
 
       class Simulator # :nodoc:
+        INITIAL_STATE = [0]
```

Comment:
Add `.freeze` to ensure the global state will be not mutated.

```suggestion
        INITIAL_STATE = [0].freeze
```